src/types/OAuthConfig.ts

```diff
@@ -11,4 +11,11 @@ type OAuthConfig = {
   queryParams?: {};
 };
 
-export { OAuthConfig };
+type TokenCacheOAuthConfig = OAuthConfig & {
+  tokenInfoEndpoint: string; // mandatory for TokenCache
```

Comment:
`tokenInfoEndpoint` can then be removed from `OAuthConfig`, it is not needed there anymore.